src/Lecture3.hs

```diff
@@ -249,4 +309,5 @@ Just [8,9,10]
 [8,20,3]
 
 -}
-apply = error "TODO"
+apply :: Applicative f => a -> f (a -> b) -> f b
+apply arg f = f <*> pure arg
```

Comment:
That's a clever solution! 👀 

The problem with this implementation is that it requires working with `Applicative`. But not every data type with the `Functor` instance has an `Applicative` instance as well. So a relaxed version of this function with the `Functor` constraint will be more generally applicable.

Reply:
Updated